src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -214,6 +214,31 @@ def mkdir(self, data):
             'zfs_attrs': ['ARCHIVE']
         }
 
+    @private
+    def listdir_request_mask(self, select):
+        """ create request mask for directory listing """
+        if not select:
+            # request_mask=None means ALL in the directory iterator
+            return None
+
+        request_mask = 0
+        for i in select:
+            selected = i[0] if isinstance(i, list) else i
+
+            match selected:
+                case 'realpath':
+                    request_mask |= DirectoryRequestMask.REALPATH
+                case 'acl':
+                    request_mask |= DirectoryRequestMask.ACL
+                case 'zfs_attrs':
+                    request_mask |= DirectoryRequestMask.ZFS_ATTRS
+                case 'is_ctldir':
+                    request_mask |= DirectoryRequestMask.CTLDIR
+                case 'xattrs':
+                    request_mask |= DirectoryRequestMask.XATTRS
+
```

Comment:
That would break for cases where we're selecting parameters that don't line up with request mask values. For example:
```
select: ["path", "xattrs"]
```